polygeist/tools/cgeist/Lib/CGCall.cc

```diff
@@ -1523,7 +1523,8 @@ ValueCategory MLIRScanner::VisitCallExpr(clang::CallExpr *expr) {
   MLIRScanner::getMangledFuncName(name, callee, Glob.CGM);
   if (isSupportedFunctions(name))
     ShouldEmit = true;
-  auto ToCall = Glob.GetOrCreateMLIRFunction(callee, ShouldEmit);
+  FunctionToEmit F{callee, mlirclang::inputContext(builder)};
+  auto ToCall = cast<func::FuncOp>(Glob.GetOrCreateMLIRFunction(F, ShouldEmit));
```

Comment:
Casting to a FuncOp does not seems generic enough.  I am thinking that you should cast based on the value of `mlirclang::inputContext(builder)`

Reply:
We cannot call SYCL kernels, so this will always be a `FuncOp` for now.

Reply:
ok